src/commands/viewProperties.ts

```diff
@@ -3,25 +3,18 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import { AppSettingsTreeItem } from 'vscode-azureappservice';
 import { IActionContext, openReadOnlyJson } from 'vscode-azureextensionui';
 import { ext } from '../extensionVariables';
 import { IAzureResourceTreeItem } from '../tree/IAzureResourceTreeItem';
 import { StaticWebAppTreeItem } from '../tree/StaticWebAppTreeItem';
-import { localize } from '../utils/localize';
-import { nonNullProp } from '../utils/nonNull';
 
-export async function viewProperties(context: IActionContext, node?: IAzureResourceTreeItem): Promise<void> {
-    if (!node) {
-        node = await ext.tree.showTreeItemPicker<StaticWebAppTreeItem>(StaticWebAppTreeItem.contextValue, context);
+export async function viewProperties(context: IActionContext, target?: IAzureResourceTreeItem | AppSettingsTreeItem): Promise<void> {
+    if (!target) {
+        target = await ext.tree.showTreeItemPicker<StaticWebAppTreeItem>(StaticWebAppTreeItem.contextValue, context);
     }
 
-    if (!node.data) {
-        if (node.getDataImpl) {
-            await node.getDataImpl();
-        } else {
-            throw new Error(localize('No data exists on resource "{0}"', node.label));
-        }
-    }
+    const data: {} = target instanceof AppSettingsTreeItem ? await target.client.listApplicationSettings() : target.data;
```

Comment:
I liked it better when this file didn't have to know about any tree item in particular. I suggest you create a simple class that extends `AppSettingsTreeItem` and implements `getDataImpl`